src/averageAges.js

```diff
@@ -54,8 +73,26 @@ function calculateWomenAverageAge(people, withChildren) {
  *
  * @return {number}
  */
-function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+function calculateAverageAgeDiff(people, onlyWithSon = false) {
+  const mothers = people.filter(person => person.sex === 'f');
+  const children = people.filter(person =>
```

Comment:
You can optimize this code by filtering mothers with children, instead of filtering all women and then filtering children.